src/components/schedule-inputs.tsx

```diff
@@ -57,6 +52,248 @@ export function ScheduleInputs(props: ScheduleInputsProps): JSX.Element | null {
+      'scheduleMonthDay'
+    ];
+
+    const newErrors = { ...props.errors };
+    for (const fieldToValidate of allFields) {
+      if (neededFields[newInterval].includes(fieldToValidate)) {
+        // Validate the field.
+        // Skip validation if value in model is undefined; this typically indicates initial load.
+        switch (fieldToValidate) {
+          case 'scheduleTime':
+            if (props.model.scheduleTimeInput !== undefined) {
+              newErrors[fieldToValidate] = validator.validateTime(
+                props.model.scheduleTimeInput
+              );
+            }
+            break;
+          case 'scheduleHourMinute':
+            if (props.model.scheduleMinuteInput !== undefined) {
+              newErrors[fieldToValidate] = validator.validateHourMinute(
+                props.model.scheduleMinuteInput
+              );
+            }
+            break;
+          case 'scheduleMonthDay':
+            if (props.model.scheduleMonthDayInput !== undefined) {
+              newErrors[fieldToValidate] = validator.validateMonthDay(
+                props.model.scheduleMonthDayInput
+              );
+            }
+            break;
+        }
+      } else {
+        // Clear validation errors.
+        newErrors[fieldToValidate] = '';
+      }
+    }
+
+    props.handleErrorsChange(newErrors);
+
+    switch (newInterval) {
+      case 'minute':
+        schedule = '* * * * *'; // every minute
+        break;
+      case 'hour':
+        schedule = `${props.model.scheduleHourMinute ?? '0'} * * * *`;
+        break;
+      case 'day':
+        schedule = `${props.model.scheduleMinute ?? '0'} ${
+          props.model.scheduleHour ?? '0'
+        } * * *`;
+        break;
+      case 'week':
+        schedule = `${props.model.scheduleMinute ?? '0'} ${
+          props.model.scheduleHour ?? '0'
+        } * * ${props.model.scheduleWeekDay ?? '1'}`;
+        dayOfWeek ??= '1'; // Default to Monday
+        break;
+      case 'weekday':
+        schedule = `${props.model.scheduleMinute ?? '0'} ${
+          props.model.scheduleHour ?? '0'
+        } * * MON-FRI`;
+        break;
+      case 'month':
+        schedule = `${props.model.scheduleMinute ?? '0'} ${
+          props.model.scheduleHour ?? '0'
+        } ${props.model.scheduleMonthDay ?? '1'} * *`;
+        break;
+    }
+
+    props.handleModelChange({
+      ...props.model,
+      schedule: schedule,
+      scheduleInterval: event.target.value,
+      scheduleWeekDay: dayOfWeek
+    });
+  };
+
+  const handleScheduleMinuteChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    let minutes = props.model.scheduleHourMinute;
+    let schedule = props.model.schedule;
+
+    const scheduleHourMinuteError = validator.validateHourMinute(value);
+
+    if (!scheduleHourMinuteError) {
+      minutes = parseInt(value);
+      // No errors; compose a new schedule in cron format
+      switch (props.model.scheduleInterval) {
```

Comment:
This `switch` pattern works in other functions in this file, but in cases like this where there is only one case, I think it's safe to assume that the one case is true. Best to remove the `switch` block and assume `…scheduleInterval === 'hour'`.

Reply:
Done. :+1: